src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -83,13 +83,18 @@ public static class TagHelperDescriptorFactory
             bool designTime)
         {
             TagHelperUsageDescriptor typeUsageDescriptor = null;
+            string outputElementHint = null;
 
-#if !DNXCORE50
             if (designTime)
             {
+                var outputElementHintAttribute =
+                    typeInfo.GetCustomAttribute<OutputElementHintAttribute>(inherit: false);
+
+                outputElementHint = outputElementHintAttribute?.OutputElement;
```

Comment:
I made this suggestion because it seemed like an unnecessary optimization that mixed the hint with descriptors and required another non-`var` variable.